pkg/ee/cluster-backup/master/sync-controller/controller.go

```diff
@@ -0,0 +1,169 @@
+      CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+      TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+      SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+   END OF TERMS AND CONDITIONS
+*/
+
+package synccontroller
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+
+	"k8c.io/kubermatic/v2/pkg/apis/equality"
+	kubermaticv1 "k8c.io/kubermatic/v2/pkg/apis/kubermatic/v1"
+	kuberneteshelper "k8c.io/kubermatic/v2/pkg/kubernetes"
+	kkpreconciling "k8c.io/kubermatic/v2/pkg/resources/reconciling"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	ControllerName = "cluster-backup-controller"
+
+	// cleanupFinalizer indicates that CBSLs on the seed clusters need cleanup.
+	cleanupFinalizer = "kubermatic.k8c.io/cleanup-seed-storage-locations"
+)
+
+type reconciler struct {
+	log          *zap.SugaredLogger
+	recorder     record.EventRecorder
+	masterClient ctrlruntimeclient.Client
+	seedClients  kuberneteshelper.SeedClientMap
+}
+
+func Add(
+	masterManager manager.Manager,
+	seedManagers map[string]manager.Manager,
+	log *zap.SugaredLogger,
+	numWorkers int,
+) error {
+	r := &reconciler{
+		log:          log.Named(ControllerName),
+		recorder:     masterManager.GetEventRecorderFor(ControllerName),
+		masterClient: masterManager.GetClient(),
+		seedClients:  kuberneteshelper.SeedClientMap{},
+	}
+
+	for seedName, seedManager := range seedManagers {
+		r.seedClients[seedName] = seedManager.GetClient()
+	}
+
+	_, err := builder.ControllerManagedBy(masterManager).
+		Named(ControllerName).
+		WithOptions(controller.Options{
+			MaxConcurrentReconciles: numWorkers,
+		}).
+		For(&kubermaticv1.ClusterBackupStorageLocation{}).
+		Build(r)
+
+	return err
+}
+
+func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	log := r.log.With("request", request)
+
+	cbsl := &kubermaticv1.ClusterBackupStorageLocation{}
+	if err := r.masterClient.Get(ctx, request.NamespacedName, cbsl); err != nil {
+		if apierrors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, fmt.Errorf("failed to get storage location: %w", err)
+	}
+
+	if !cbsl.DeletionTimestamp.IsZero() {
+		if err := r.handleDeletion(ctx, log, cbsl); err != nil {
+			return reconcile.Result{}, fmt.Errorf("handling deletion: %w", err)
+		}
+		return reconcile.Result{}, nil
+	}
+
+	cbslReconcilerFactories := []kkpreconciling.NamedClusterBackupStorageLocationReconcilerFactory{
+		cbslReconcilerFactory(cbsl),
+	}
+
+	err := r.seedClients.Each(ctx, log, func(_ string, seedClient ctrlruntimeclient.Client, log *zap.SugaredLogger) error {
```

Comment:
nit: I think it might be useful to actually show the seed cluster name in the logs here. It would simplify debugging if there are issues. Similarly with the delete call.